python-sdk/tutorials/automl-with-azureml/image-object-detection/ONNX_batch_model_generator_faster_rcnn.py

```diff
@@ -0,0 +1,126 @@
+                      dynamic_axes=dynamic_axes)
+
+def generate_onnx_batch_model(batch_size, height_onnx, width_onnx, task_type, best_child_run, model_settings):
+    """
+    Get the onnx batch model and verify the schema.
+
+    :param batch_size: Batch size needed for batch inference
+    :type batch_size: int
+    :param height_onnx: Height of the image to be used for ONNX model
+    :type height_onnx: int
+    :param width_onnx: Width of the image to be used for ONNX model
+    :type width_onnx: int
+    :param task_type: Type of vision task from image classificatio, object detection, segmentation
+    :type task_type: str
+    :param best_child_run: best child run object of an experiment
+    :type best_child_run: azureml.core.run.Run
+    :param model_settings: model parameters with key, values
+    :type model_settings: dict
+    """
+
+    # download the pytorch model weights
+    torch_model_path = "model.pt"
+    best_child_run.download_file(name="train_artifacts/model.pt", output_file_path=torch_model_path)
+    # load the model wrapper
+    model_wrapper = load_model(task_type, torch_model_path, **model_settings)
+
+    if (batch_size > 1) or (height_onnx != 600) or (width_onnx != 800):
+        onnx_model_file_path = "./outputs/model_" + str(batch_size) + ".onnx"
+        export_onnx_model(model_wrapper, batch_size, height_onnx, width_onnx, onnx_model_file_path, device='cpu')
+    else:
+        msg = "Please use the auto-generated ONNX model for the best child run. No need to run this script"
+        logger.warning(msg)
+    try:
+        # check/verify schema of generated onnx model
+        onnx_model = onnx.load(onnx_model_file_path)
+        onnx.checker.check_model(onnx_model)
+        logger.info("ONNX model generation for the batch size {} is successful".format(batch_size))
+    except Exception as e:
+        logger.error("ONNX model Schema validation error: " + str(e))
+        raise
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    # add arguments
+    parser.add_argument('--batch_size', type=int, required=True)
+    parser.add_argument('--height_onnx', type=int, required=True)
+    parser.add_argument('--width_onnx', type=int, required=True)
+    parser.add_argument('--experiment_name', type=str, required=True)
+    parser.add_argument('--subscription_id', type=str, required=True)
+    parser.add_argument('--resource_group', type=str, required=True)
+    parser.add_argument('--workspace_name', type=str, required=True)
+    parser.add_argument('--run_id', type=str, required=True)
+    parser.add_argument('--task_type', type=str, required=True)
+    parser.add_argument('--min_size', type=int, required=True)
+    parser.add_argument('--max_size', type=int, required=True)
+    parser.add_argument('--box_score_thresh', type=float, required=True)
+    parser.add_argument('--box_nms_thresh', type=float, required=True)
+    parser.add_argument('--box_detections_per_img', type=int, required=True)
```

Comment:
why do these matter for the onnx export?

Reply:
they needed for this step:
model_wrapper = load_model(task_type, torch_model_path, **model_settings)